homeassistant/components/dsmr_reader/definitions.py

```diff
@@ -119,6 +121,11 @@ def tariff_transform(value):
         "icon": "mdi:flash",
         "unit": ELECTRICAL_CURRENT_AMPERE,
     },
+    "dsmr/reading/timestamp": {
+        "name": "Telegram timestamp",
+        "icon": "mdi:clock",
+        "unit": DEVICE_CLASS_TIMESTAMP,
```

Comment:
This is actually expected a unit of measurement, not a device class.

Reply:
Ah yes.. I've added the device_class and changed the unit_of_measurement with fallback on device_class.

Reply:
The timestamp device class should not set a unit. Just return `None` as unit in that case.

Please show us data for the usefulness of this sensor first, before working more here.

Reply:
Should be fixed now!